crystalfontz/dbus/map.py

```diff
@@ -0,0 +1,580 @@
+
+    @staticmethod
+    def load(unit: bool) -> TemperatureUnit:
+        return TEMPERATURE_UNITS[unit]
+
+    @staticmethod
+    def dump(unit: TemperatureUnit) -> bool:
+        return bool(unit.value)
+
+
+class TemperatureDisplayItemM:
+    t: ClassVar[str] = t(IndexM, TemperatureDigitsM, PositionM, PositionM, "b")
+
+    @staticmethod
+    def load(
+        item: Tuple[int, int, int, int, bool],
+    ) -> TemperatureDisplayItem:
+
+        index, n_digits, column, row, units = item
+        return TemperatureDisplayItem(
+            index,
+            TemperatureDigitsM.load(n_digits),
+            column,
+            row,
+            TemperatureUnitM.load(units),
+        )
+
+    @staticmethod
+    def dump(item: TemperatureDisplayItem) -> Tuple[int, int, int, int, bool]:
+        return (
+            item.index,
+            item.n_digits,
+            item.column,
+            item.row,
+            TemperatureUnitM.dump(item.units),
+        )
+
+
+class SetupLiveTemperatureDisplayM:
+    t: ClassVar[str] = t(IndexM, TemperatureDisplayItemM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        slot: int,
+        item: Tuple[int, int, int, int, bool],
+        timeout: float,
+        retry_times: int,
+    ) -> Tuple[int, TemperatureDisplayItem, Optional[float], Optional[int]]:
+        return (
+            slot,
+            TemperatureDisplayItemM.load(item),
+            TimeoutM.load(timeout),
+            RetryTimesM.load(retry_times),
+        )
+
+
+LCD_REGISTERS: Dict[bool, LcdRegister] = {
+    bool(register.value): register for register in LcdRegister
+}
+
+
+class LcdRegisterM:
+    t: ClassVar[str] = "b"
+
+    @staticmethod
+    def load(register: bool) -> LcdRegister:
+        return LCD_REGISTERS[register]
+
+    @staticmethod
+    def dump(register: LcdRegister) -> bool:
+        return bool(register.value)
+
+
+class SendCommandToLcdControllerM:
+    t: ClassVar[str] = t(LcdRegisterM, ByteM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        location: bool, data: int, timeout: float, retry_times: int
+    ) -> Tuple[LcdRegister, int, Optional[float], Optional[int]]:
+        return (
+            LcdRegisterM.load(location),
+            data,
+            TimeoutM.load(timeout),
+            RetryTimesM.load(retry_times),
+        )
+
+
+class ConfigureKeyReportingM:
+    t: ClassVar[str] = t(array(ByteM), array(ByteM), TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        when_pressed: List[int],
+        when_released: List[int],
+        timeout: float,
+        retry_times: int,
+    ) -> Tuple[Set[int], Set[int], Optional[float], Optional[int]]:
+        return (
+            set(when_pressed),
+            set(when_released),
+            TimeoutM.load(timeout),
+            RetryTimesM.load(retry_times),
+        )
+
+
+class SpecialCharacterM:
+    t: ClassVar[str] = "t"
+
+    @staticmethod
+    def load(character: int) -> SpecialCharacter:
```

Comment:
My intent here is to map a `uint64` to `List[List[bool]]` and construct a special charater from that.